eng/pipelines/runtime.yml

```diff
@@ -208,6 +208,7 @@ jobs:
     - Browser_wasm
     - tvOS_arm64
     - iOS_arm64
+    - MacCatalyst_x64
```

Comment:
nit: if I understand this correctly this job builds the AOT offsets for _all_ archs, not just `iOS_arm64` or `MacCatalyst_x64` right? we should add a comment to explain it since this is quite confusing.